src/rpc/server.cpp

```diff
@@ -235,12 +236,44 @@ static UniValue getrpcinfo(const JSONRPCRequest& request)
     return result;
 }
 
+static UniValue getrpcwhitelist(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"getrpcwhitelist",
+                "\nReturns whitelisted RPCs for the current user.\n",
+                {},
+                RPCResult{
+                    RPCResult::Type::OBJ, "", "",
+                    {
+                        {RPCResult::Type::ARR, "methods", "List of RPCs that the user is allowed to call",
```

Comment:
Is there a case for making it an Object with possible details of more fine-grained calls? Or is it safe to assume we won't go beyond the method level for permissions?

Reply:
I think that's a decent idea for future-proofing the return type, but there is some disagreement of the level of permission control we want to allow in core v.s. a layer on top.